File: audit.js
const fs = require('fs');
const path = require('path');

const ROOT = process.cwd();
const SRC = path.join(ROOT, 'src');

function findFiles(dir, exts = ['.tsx', '.ts']) {
  const results = [];
  if (!fs.existsSync(dir)) return results;
  const items = fs.readdirSync(dir, { withFileTypes: true });
  for (const item of items) {
    const full = path.join(dir, item.name);
    if (item.isDirectory()) {
      results.push(...findFiles(full, exts));
    } else if (exts.some(e => item.name.endsWith(e))) {
      results.push(full);
    }
  }
  return results;
}

const allFiles = findFiles(SRC);

const issues = {
  next_link: [],
  next_nav: [],
  missing_components: [],
  env_vars: new Set(),
  supabase_tables: new Set(),
  todo_fixme: [],
  hardcoded_text: [],
  wrong_supabase: [],
};

for (const f of allFiles) {
  const text = fs.readFileSync(f, 'utf8');
  const lines = text.split('\n');
  const rel = f.replace(ROOT + path.sep, '').replace(/\\/g, '/');

  lines.forEach((line, i) => {
    const ln = i + 1;

    // Non-localized imports
    if ((line.includes("from 'next/link'") || line.includes('from "next/link"')) && !line.includes('//')) {
      issues.next_link.push(`${rel}:${ln}: ${line.trim().slice(0, 80)}`);
    }
    if ((line.includes("from 'next/navigation'") || line.includes('from "next/navigation"')) && !line.includes('//')) {
      if (line.includes('useRouter') || line.includes('usePathname') || line.includes('redirect')) {
        issues.next_nav.push(`${rel}:${ln}: ${line.trim().slice(0, 80)}`);
      }
    }

    // Env vars
    const envMatches = line.match(/process\.env\.[A-Z_]+/g) || [];
    envMatches.forEach(m => issues.env_vars.add(m));

    // Supabase tables
    const tableMatches = line.match(/\.from\('([^']+)'\)/g) || [];
    tableMatches.forEach(m => {
      const t = m.match(/\.from\('([^']+)'\)/);
      if (t) issues.supabase_tables.add(t[1]);
    });

    // TODO/FIXME
    if (/\b(TODO|FIXME)\b/i.test(line)) {
      issues.todo_fixme.push(`${rel}:${ln}: ${line.trim().slice(0, 80)}`);
    }

    // Hardcoded IT text (not in message files)
    if (!rel.includes('/messages/') && /Centra su di me|coming soon|lorem ipsum|PLACEHOLDER/i.test(line)) {
      issues.hardcoded_text.push(`${rel}:${ln}: ${line.trim().slice(0, 80)}`);
    }

    // Wrong Supabase URL
    if (line.toLowerCase().includes('kvmdeolhkpnhkomnhlsy')) {
      issues.wrong_supabase.push(`${rel}:${ln}`);
    }
  });

  // Component import check (only for app pages)
  if (rel.startsWith('src/app')) {
    const imports = text.match(/from ['"](@\/components\/[^'"]+)['"]/g) || [];
    imports.forEach(imp => {
      const compPath = imp.match(/from ['"](@\/components\/[^'"]+)['"]/)[1];
      const localPath = compPath.replace('@/', 'src/');
      const candidates = [
        path.join(ROOT, localPath + '.tsx'),
        path.join(ROOT, localPath + '.ts'),
        path.join(ROOT, localPath, 'index.tsx'),
        path.join(ROOT, localPath, 'index.ts'),
      ];
      if (!candidates.some(c => fs.existsSync(c))) {
        const key = `${compPath} (from ${rel})`;
        if (!issues.missing_components.includes(key)) {
          issues.missing_components.push(key);
        }
      }
    });
  }
}

// Check .env.local
const envLocal = path.join(ROOT, '.env.local');
const envContent = fs.existsSync(envLocal) ? fs.readFileSync(envLocal, 'utf8') : '';
const requiredEnv = [
  'NEXT_PUBLIC_SUPABASE_URL',
  'NEXT_PUBLIC_SUPABASE_ANON_KEY',
  'NEXT_PUBLIC_MAPBOX_TOKEN',
  'STRIPE_SECRET_KEY',
  'CRON_SECRET',
];
const missingEnv = requiredEnv.filter(v => !envContent.includes(v));

// Check page files
const appDir = path.join(SRC, 'app', '[locale]');
const appRoutes = [];
if (fs.existsSync(appDir)) {
  fs.readdirSync(appDir, { withFileTypes: true }).forEach(d => {
    if (d.isDirectory()) {
      const pg = path.join(appDir, d.name, 'page.tsx');
      appRoutes.push({ route: d.name, exists: fs.existsSync(pg) });
    }
  });
}

// REPORT
console.log('='.repeat(65));
console.log('VIBEMEET FULL AUDIT REPORT');
console.log('='.repeat(65));

console.log(`\n[1] NON-LOCALIZED next/link (${issues.next_link.length} files):`);
issues.next_link.forEach(x => console.log('  ' + x));

console.log(`\n[2] NON-LOCALIZED next/navigation (${issues.next_nav.length} occurrences):`);
// Group by file
const navFiles = [...new Set(issues.next_nav.map(x => x.split(':')[0]))];
navFiles.forEach(f => console.log('  ' + f));

console.log(`\n[3] MISSING COMPONENTS (${issues.missing_components.length}):`);
issues.missing_components.forEach(x => console.log('  ' + x));

console.log(`\n[4] ENV VARS USED (${issues.env_vars.size}):`);
[...issues.env_vars].sort().forEach(x => console.log('  ' + x));

console.log(`\n[5] MISSING ENV VARS in .env.local (${missingEnv.length}):`);
missingEnv.forEach(x => console.log('  !! ' + x));

console.log(`\n[6] SUPABASE TABLES (${issues.supabase_tables.size}):`);
[...issues.supabase_tables].sort().forEach(x => console.log('  ' + x));

console.log(`\n[7] TODO/FIXME (${issues.todo_fixme.length}):`);
issues.todo_fixme.forEach(x => console.log('  ' + x));

console.log(`\n[8] HARDCODED TEXT in pages (${issues.hardcoded_text.length}):`);
issues.hardcoded_text.forEach(x => console.log('  ' + x));

console.log(`\n[9] WRONG SUPABASE URL (${issues.wrong_supabase.length}):`);
issues.wrong_supabase.forEach(x => console.log('  ' + x));

console.log(`\n[10] APP ROUTES CHECK (${appRoutes.length} dirs):`);
appRoutes.filter(r => !r.exists).forEach(r => console.log(`  MISSING page.tsx: ${r.route}`));
const missing = appRoutes.filter(r => !r.exists);
if (missing.length === 0) console.log('  All route page.tsx files exist ✓');

console.log('\n' + '='.repeat(65));
console.log('SUMMARY:');
console.log(`  BUILD: PASS (exit 0)`);
console.log(`  next/link misuse: ${issues.next_link.length}`);
console.log(`  next/navigation misuse: ${navFiles.length} files`);
console.log(`  Missing components: ${issues.missing_components.length}`);
console.log(`  Missing env vars: ${missingEnv.length}`);
console.log(`  Supabase tables: ${issues.supabase_tables.size}`);
console.log(`  TODO/FIXME: ${issues.todo_fixme.length}`);
console.log(`  Hardcoded text in pages: ${issues.hardcoded_text.length}`);
console.log(`  Wrong Supabase URL: ${issues.wrong_supabase.length}`);
console.log('='.repeat(65));

File: audit.py
#!/usr/bin/env python3
import os, re, json
from pathlib import Path

ROOT = Path(".")
SRC = ROOT / "src"

def find_files(exts=(".tsx",".ts")):
    for ext in exts:
        yield from SRC.rglob(f"*{ext}")

results = {
    "next_link_misuse": [],
    "next_navigation_misuse": [],
    "missing_components": [],
    "env_vars": set(),
    "supabase_tables": set(),
    "console_errors": [],
    "todo_fixme": [],
    "hardcoded_text": [],
    "hardcoded_supabase_url": [],
}

# Known localized imports that are OK
OK_PATHS = ["i18n/navigation", "lib/i18n"]

for f in find_files():
    text = f.read_text(encoding="utf-8", errors="ignore")
    lines = text.splitlines()
    rel = str(f).replace("\\", "/")

    for i, line in enumerate(lines, 1):
        # 1. Non-localized next/link or next/navigation
        if ("from 'next/link'" in line or 'from "next/link"' in line):
            results["next_link_misuse"].append(f"{rel}:{i}: {line.strip()}")
        if ("from 'next/navigation'" in line or 'from "next/navigation"' in line):
            results["next_navigation_misuse"].append(f"{rel}:{i}: {line.strip()}")

        # 2. Env vars
        for m in re.finditer(r"process\.env\.[A-Z_]+", line):
            results["env_vars"].add(m.group())

        # 3. Supabase tables
        for m in re.finditer(r"\.from\('([^']+)'\)", line):
            results["supabase_tables"].add(m.group(1))

        # 4. Console errors
        if "console.error" in line or "console.warn" in line:
            results["console_errors"].append(f"{rel}:{i}")

        # 5. TODO/FIXME
        if re.search(r"\bTODO\b|\bFIXME\b", line, re.I):
            results["todo_fixme"].append(f"{rel}:{i}: {line.strip()[:80]}")

        # 6. Hardcoded text patterns (Centra su di me, coming soon, etc.)
        if re.search(r"Centra su di me|coming soon|lorem ipsum|PLACEHOLDER|hardcoded", line, re.I):
            results["hardcoded_text"].append(f"{rel}:{i}: {line.strip()[:80]}")

        # 7. Wrong Supabase URL
        if "kvmdeolhkpnhkomnhlsy" in line.lower():
            results["hardcoded_supabase_url"].append(f"{rel}:{i}: {line.strip()[:80]}")

# Check component imports from app pages
import_pattern = re.compile(r"from ['\"](@/components/[^'\"]+)['\"]")
all_imports = set()
for f in (SRC / "app").rglob("*.tsx"):
    text = f.read_text(encoding="utf-8", errors="ignore")
    for m in import_pattern.finditer(text):
        all_imports.add(m.group(1))

for imp in sorted(all_imports):
    # Convert @/components/foo/Bar to src/components/foo/Bar
    local = imp.replace("@/", "src/")
    candidates = [
        Path(local + ".tsx"),
        Path(local + ".ts"),
        Path(local + "/index.tsx"),
        Path(local + "/index.ts"),
    ]
    if not any(c.exists() for c in candidates):
        results["missing_components"].append(imp)

# Check .env.local exists and contains required vars
required_env = [
    "NEXT_PUBLIC_SUPABASE_URL",
    "NEXT_PUBLIC_SUPABASE_ANON_KEY",
    "NEXT_PUBLIC_MAPBOX_TOKEN",
    "STRIPE_SECRET_KEY",
    "CRON_SECRET",
]
env_local = ROOT / ".env.local"
env_content = env_local.read_text(encoding="utf-8", errors="ignore") if env_local.exists() else ""
missing_env = [v for v in required_env if v not in env_content]

# Print report
print("=" * 60)
print("VIBEMEET FULL AUDIT REPORT")
print("=" * 60)

print(f"\n[1] NON-LOCALIZED next/link imports ({len(results['next_link_misuse'])}):")
for x in results["next_link_misuse"][:20]:
    print(f"  {x}")

print(f"\n[2] NON-LOCALIZED next/navigation imports ({len(results['next_navigation_misuse'])}):")
for x in results["next_navigation_misuse"][:20]:
    print(f"  {x}")

print(f"\n[3] MISSING COMPONENTS ({len(results['missing_components'])}):")
for x in results["missing_components"]:
    print(f"  {x}")

print(f"\n[4] ENV VARS USED ({len(results['env_vars'])}):")
for x in sorted(results["env_vars"]):
    print(f"  {x}")

print(f"\n[5] MISSING ENV VARS in .env.local ({len(missing_env)}):")
for x in missing_env:
    print(f"  {x}")

print(f"\n[6] SUPABASE TABLES QUERIED ({len(results['supabase_tables'])}):")
for x in sorted(results["supabase_tables"]):
    print(f"  {x}")

print(f"\n[7] TODO/FIXME ({len(results['todo_fixme'])}):")
for x in results["todo_fixme"][:20]:
    print(f"  {x}")

print(f"\n[8] HARDCODED TEXT ({len(results['hardcoded_text'])}):")
for x in results["hardcoded_text"][:20]:
    print(f"  {x}")

print(f"\n[9] WRONG SUPABASE URL ({len(results['hardcoded_supabase_url'])}):")
for x in results["hardcoded_supabase_url"]:
    print(f"  {x}")

print(f"\n[10] FILES WITH console.error/warn: {len(set(x.split(':')[0] for x in results['console_errors']))}")

print("\n" + "=" * 60)
print("SUMMARY:")
print(f"  next/link misuse: {len(results['next_link_misuse'])}")
print(f"  next/navigation misuse: {len(results['next_navigation_misuse'])}")
print(f"  Missing components: {len(results['missing_components'])}")
print(f"  Missing env vars: {len(missing_env)}")
print(f"  TODO/FIXME: {len(results['todo_fixme'])}")
print(f"  Hardcoded text: {len(results['hardcoded_text'])}")
print(f"  Wrong Supabase URL: {len(results['hardcoded_supabase_url'])}")
print("=" * 60)
